### PR TITLE
feat: post error messages on slack

### PR DESCRIPTION
**Pull request by amateima**
- add `alerting` module and [AlertingService](https://github.com/across-protocol/indexer/pull/105/files#diff-1bc16a621534659f7f89635243825fd17b6007f0f959d54a2031b7cb8e44e266R5) class to post messages to Slack
- plug the `alerting` package in all `Indexer` and `BaseIndexer` classes 

Screenshot 2024-11-19 at 14 00 17

### CODE REVIEW

File: packages/alerting/src/service/AlertingService.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import axios from "axios";
+import { AlertingConfig, SlackMessagePayload } from "../model";
+
+export class AlertingService {
+  constructor(
+    private config: AlertingConfig,
+    private logger: Logger,
+  ) {}
+
+  public async postMessageOnSlack(payload: SlackMessagePayload) {
```
Comment (daywiss) [inline packages/alerting/src/service/AlertingService.ts:11]: dont we have slack loggers in winston?

**Review by daywiss — COMMENTED** [review]
i thought we had ability to post in slack through some kind of winston plugin, but maybe im mistaken

File: packages/indexer/src/services/BundleIncludedEventsService.ts [hunk 2]
```diff
@@ -24,35 +27,28 @@ export type BundleConfig = {
...
-        at: "BundleIncludedEventsService#indexerLogic",
-        message: "Finished BundleIncludedEventsService",
-      });
-    } catch (error) {
-      this.logger.error({
-        at: "BundleIncludedEventsService#indexerLogic",
-        message: "Error in BundleIncludedEventsService",
-        error,
-      });
-    }
```
Comment (melisaguevara) [inline packages/indexer/src/services/BundleIncludedEventsService.ts:55]: Just OOC, why are we removing this try/catch block?

**Review by melisaguevara — APPROVED** [review]
looks good! Just one question

File: packages/alerting/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+```ts
+const alertingService = new AlertingService({
+    slack: {
+        enabled: true | false,
+        webhookUrl: <SLACK_WEBHOOK_URL>
+    }
+}, logger);
+await alertingService.postMessageOnSlack({
+    header: "My custom header",
+    messages: ["My custom paragraph 1", "My custom paragraph 2"]
```
Comment (james-a-morris) [inline packages/alerting/README.md:16]: is this a copy/paste from somewhere? Do slack posts lose the `{ at: ..., message: ..., ...}`

File: packages/alerting/main.test.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { expect } from "chai";
+
+describe("main", () => {
+  it("should return true", async () => {
+    expect(true).to.be.true;
+  });
```
Comment (james-a-morris) [inline packages/alerting/main.test.ts:6]: Should we start adding tests?

File: packages/alerting/src/service/AlertingService.ts (hunk 1 above)
Comment (james-a-morris) [inline packages/alerting/src/service/AlertingService.ts:11]: Agreed on David's post - this has been done in UMA logging with most of the try/catch/paging handling internally

**Comment by james-a-morris** [comment]
https://github.com/across-protocol/indexer/pull/105#discussion_r1848858457

**Pull request closed without merging**